#--- syslog-ng-3.5.2/configure.ac~	2013-12-01 22:52:03.000000000 +0100
#+++ syslog-ng-3.5.2/configure.ac	2013-12-01 23:02:32.955209559 +0100
#@@ -1145,7 +1145,7 @@
# else
# 	SYSLOGNG_DEPS_LIBS="$LIBS $BASE_LIBS $RESOLV_LIBS $LD_START_STATIC -Wl,--whole-archive $GLIB_LIBS $EVTLOG_LIBS $PCRE_LIBS $REGEX_LIBS -Wl,--no-whole-archive $IVYKIS_NO_LIBTOOL_LIBS $LD_END_STATIC $LIBCAP_LIBS $DL_LIBS "
# 	TOOL_DEPS_LIBS="$LIBS $BASE_LIBS $GLIB_LIBS $EVTLOG_LIBS $RESOLV_LIBS $LIBCAP_LIBS $PCRE_LIBS $REGEX_LIBS $IVYKIS_LIBS $DL_LIBS"
#-	CORE_DEPS_LIBS=""
#+	CORE_DEPS_LIBS="$GLIB_LIBS $EVTLOG_LIBS $LIBCAP_LIBS $IVYKIS_LIBS"
# 
# 	# bypass libtool in case we want to do mixed linking because it
# 	# doesn't support -Wl,-Bstatic -Wl,-Bdynamic on a per-library basis.
--- syslog-ng-3.6.3/syslog-ng-ctl/Makefile.am.orig	2015-06-08 15:44:04.000000000 +0200
+++ syslog-ng-3.6.3/syslog-ng-ctl/Makefile.am	2015-06-14 12:14:41.989973064 +0200
@@ -10,5 +10,6 @@ syslog_ng_ctl_syslog_ng_ctl_SOURCES		=
 EXTRA_DIST					+=	\
 	syslog-ng-ctl/control-client-unix.c
 
-syslog_ng_ctl_syslog_ng_ctl_LDADD		= lib/libsyslog-ng.la @BASE_LIBS@ @GLIB_LIBS@ @RESOLV_LIBS@
-
+syslog_ng_ctl_syslog_ng_ctl_DEPENDENCIES	= lib/libsyslog-ng.la
+syslog_ng_ctl_syslog_ng_ctl_LDADD		= -L${top_builddir}/lib/.libs -lsyslog-ng @SYSLOGNG_DEPS_LIBS@
+syslog_ng_ctl_syslog_ng_ctl_LINK		= @SYSLOGNG_LINK@
--- syslog-ng-3.6.3/lib/logproto/tests/Makefile.am.orig	2015-06-08 15:44:04.000000000 +0200
+++ syslog-ng-3.6.3/lib/logproto/tests/Makefile.am	2015-06-14 12:29:25.672597931 +0200
@@ -20,7 +20,7 @@
 
 lib_logproto_tests_test_findeom_CFLAGS	= -I${top_srcdir}/libtest
 lib_logproto_tests_test_findeom_LDADD	= \
-	${top_builddir}/lib/libsyslog-ng.la \
+	$(TEST_LDADD) \
 	${top_builddir}/libtest/libsyslog-ng-test.a
 lib_logproto_tests_test_findeom_SOURCES = \
 	lib/logproto/tests/test_findeom.c
--- syslog-ng-3.8.1/modules/diskq/Makefile.am~	2016-10-29 16:04:03.000000000 +0200
+++ syslog-ng-3.8.1/modules/diskq/Makefile.am	2016-10-29 16:04:53.759617687 +0200
@@ -45,9 +45,9 @@
 modules_diskq_libdisk_buffer_la_DEPENDENCIES = $(MODULE_DEPS_LIBS) $(LIBSYSLOG_NG_DISK_BUFFER)
 
 modules_diskq_dqtool_SOURCES = modules/diskq/dqtool.c
 modules_diskq_dqtool_LDADD = \
-  $(TOOL_DEPS_LIBS) \
   $(MODULE_DEPS_LIBS) \
+  $(TOOL_DEPS_LIBS) \
   $(LIBSYSLOG_NG_DISK_BUFFER)
 
 modules/diskq modules/diskq/ mod-diskq: modules/diskq/libdisk-buffer.la \
